### PR TITLE
DS Prefill: Kimi K2.6 dequantization support

### LINKED ISSUE #48449: Support downloading and dequantization of Kimi weights
Kimi K2.6 weights on HF are in INT4 format, which is different compared to DeepSeek v3. Currently, the mechanism for downloading model weights for DS Prefill exists, but the support for their dequantization is missing.

The goal is to add support for their dequantization so weights can be 'automagically' downloaded, dequantized and used by the model.

### PR DESCRIPTION
**Pull request by nbabinTT**
### PR Category
Feature

Closes https://github.com/tenstorrent/tt-metal/issues/48449

### Summary
Adds Kimi K2.6 weight-dequantization support to the DS-prefill (`deepseek_v3_d_p`) test pipeline so the prefill block/transformer can run against real Kimi K2.6 pretrained weights.

Kimi K2.6 ships as a multimodal `compressed-tensors` checkpoint whose routed-expert weights are **pack-quantized INT4** (not DeepSeek's fp8 block-wise scheme), with the language model nested under a `language_model.` key prefix. Two problems followed from that:

1. The shared `deepseek_v3` dequantizer only understands fp8 block-wise weights, so Kimi's INT4 experts couldn't be loaded.
2. Every weight-addressing call assumed bare `model.`/`lm_head.` keys, which don't exist in the raw multimodal checkpoint.

This PR solves both with a small, self-contained, pure-torch path and no new dependency on an external quant library.

**What's added**

- **Pure-torch INT4 dequantizer** (`deepseek_v3_d_p/utils/test_utils.py`): unpacks `compressed-tensors` pack-quantized symmetric group-wise INT4 weights (`weight_packed`/`weight_scale`/`weight_shape` triplets → dense bf16), the INT counterpart to the shared fp8 `dequantize_weight_tensor`. Non-quantized tensors pass through with their source dtype preserved (notably the fp32 `e_score_correction_bias` feeding the router top-k, which must not be downcast). Unsupported layouts (asymmetric, non-group strategy, activation reordering) raise `NotImplementedError` rather than silently mis-dequantizing.
- **d_p-local `dequantize_state_dict`**: routes pack-quantized INT4 checkpoints to the new local path and delegates everything else (DeepSeek fp8) unchanged to the shared `deepseek_v3` dequantizer.
- **`detect_language_model_prefix`**: detects the `language_model.` nesting from the actual keys, so the same variant works for either a raw multimodal checkpoint or an already-dequantized/stripped one. Wired through `pretrained_transformer_weights` (conftest) and `load_and_compute_layer_by_layer` (transformer_helpers).

**Test wiring**

- Promotes `use_pretrained` to an explicit `random`/`pretrained` parametrize axis on `test_ds_prefill_block` and `test_kimi_prefill_block`, replacing the implicit "checkpoint-present → use real weights" gate. `model_path`/`weight_cache_path` are now resolved lazily (via `request.getfixturevalue`) only on the pretrained path, so the random path never triggers a multi-GB HuggingFace download. Removes the now-redundant block-scoped `model_path` fixture override and the `weight_block_size` stubbing hack in `_unwrap_multimodal_config`.

**CI**

- Adds `and not pretrained` to the DS-prefill legs in `blackhole_e2e_tests.yaml` and `demo_sp_release_tests.yaml` so the new axis doesn't double those legs (and avoids strict XPASS on machines without local weights).
- Adds `and pretrained` to the Galaxy DS-prefill legs that should exercise real weights, and wires the Kimi dequantized checkpoint + TTNN cache env vars, adding a 25k-prompt Kimi prefill-block leg alongside the existing 5k.

### Notes for reviewers

- Core review surface is the INT4 unpacking math in `_dequantize_packed_int4_weight` (little-endian lane layout, offset-binary → signed levels, per-group scale broadcast) — the docstring documents the exact `compressed-tensors` layout it assumes.
- The `and not pretrained` / `and pretrained` `-k` filter edits change which legs run; worth a sanity check that the pretrained legs land on machines that actually have the checkpoints mounted.

### DeepSeek Prefill CI

- [ ] [(Blackhole) e2e tests](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-e2e-tests.yaml?query=branch:nbabin/kimi_dequant)
- [ ] [(Galaxy) DeepSeek Prefill tests](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-deepseek-prefill-tests.yaml?query=branch:nbabin/kimi_dequant)
- [ ] [(T3K) T3000 e2e tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-e2e-tests.yaml?query=branch:nbabin/kimi_dequant) (same error as on main)
- [ ] [(Release) Model Status for Console Deployment](https://github.com/tenstorrent/tt-metal/actions/workflows/demo-sp-release.yaml?query=branch:nbabin/kimi_dequant)

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:nbabin/kimi_dequant)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:nbabin/kimi_dequant)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-sanity-tests.yaml?query=branch:nbabin/kimi_dequant)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:nbabin/kimi_dequant)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:nbabin/kimi_dequant)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:nbabin/kimi_dequant)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:nbabin/kimi_dequant)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:nbabin/kimi_dequant)

### CODE REVIEW

File: models/demos/deepseek_v3_d_p/utils/test_utils.py [hunk 1]
```diff
@@ -115,3 +117,170 @@ def save_intermediate_output(
...
+def _dequantize_packed_int4_weight(
+    packed: torch.Tensor,
+    scale: torch.Tensor,
+    shape: torch.Tensor | Sequence[int],
+    *,
+    group_size: int,
+    num_bits: int = 4,
+    symmetric: bool = True,
+    dtype: torch.dtype = torch.bfloat16,
+) -> torch.Tensor:
```
[the remark this answers is not in the document]
Comment (ddjekicTT) [reply models/demos/deepseek_v3_d_p/utils/test_utils.py:198]: agreed with this one @nbabinTT

[the remark this answers is not in the document]
Comment (ddjekicTT) [reply models/demos/deepseek_v3_d_p/utils/test_utils.py:198]: something like `test_dequantize_state_dict_compat_shim_handles_quantized_inputs()` in `tt-metal/models/demos/deepseek_v3/tests/test_hf_model_utils.py`

**Review by ddjekicTT — APPROVED** [review]
looks good, please address the comments

File: models/demos/deepseek_v3_d_p/utils/test_utils.py [hunk 2]
```diff
@@ -115,3 +117,170 @@ def save_intermediate_output(
...
+        return False
+    for group in (quantization_config.get("config_groups") or {}).values():
+        if ((group or {}).get("weights") or {}).get("num_bits") == 4:
+            return True
+    return False
+
+
+def _pack_quant_params(quantization_config: dict) -> tuple[int, int, bool]:
+    """Extract and validate ``(num_bits, group_size, symmetric)`` for the pack-quantized weight group."""
+    weights_cfg = quantization_config["config_groups"]["group_0"]["weights"]
```
[the remark this answers is not in the document]
Comment (nbabinTT) [reply models/demos/deepseek_v3_d_p/utils/test_utils.py]: done

File: models/demos/deepseek_v3_d_p/utils/test_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nbabinTT) [reply models/demos/deepseek_v3_d_p/utils/test_utils.py:198]: done, thank you!

**Review by tdowdallTT — APPROVED** [review]
